crates/bevy_ui/src/node_bundles.rs

```diff
@@ -105,6 +108,47 @@ pub struct ImageBundle {
     pub z_index: ZIndex,
 }
 
+/// A UI node that is a texture atlas sprite
+#[derive(Bundle, Debug, Default)]
+pub struct TextureAtlasImageBundle {
+    /// Describes the logical size of the node
+    ///
+    /// This field is automatically managed by the UI layout system.
+    /// To alter the position of the `NodeBundle`, use the properties of the [`Style`] component.
+    pub node: Node,
+    /// Styles which control the layout (size and position) of the node and it's children
+    /// In some cases these styles also affect how the node drawn/painted.
+    pub style: Style,
+    /// The calculated size based on the given image
+    pub calculated_size: ContentSize,
+    /// The background color, which serves as a "fill" for this node
+    ///
+    /// Combines with `UiImage` to tint the provided image.
+    pub background_color: BackgroundColor,
+    /// A handle to the texture atlas to use for this Ui Node
+    pub texture_atlas: Handle<TextureAtlas>,
+    /// The descriptor for which sprite to use from the given texture atlas
+    pub texture_atlas_sprite: UiTextureAtlasSprite,
```

Comment:
Ah I understand, I didn't realise you'd done it that. I'm not sure how it's useful though as we only need the `flip_x`, `flip_y` and `index` fields.

Reply:
Good point, I'll just use the fields we need and create it as a new standalone struct.  My dream was to reuse as much of the existing infrastructure as possible but I'm reconsidering that now.

Reply:
I still think `UiTextureAtlasSprite` has misleading semantics. Otherwise, everything else is looking good.

Reply:
Can you elaborate on what you mean by misleading semantics here? I think mirroring the existing texture atlas mechanics and names is what I would expect (outside of ui this is the same name and fields just here I've added Ui to the front of the name)

Reply:
The term "sprite" typically refers to graphics in 2D games representing in-game objects that can be positioned freely.    So I think it's confusing to use it use to it to describe an image that is part of a UI layout where its position and size are derived from the style components by the layout algorithm, rather than set directly. 

Also the names `UiImage` and `UiTextureAtlasSprite` suggest more substantial differences between the two than just the implementation. It's more consistent to use the same terminology for both, either "sprite" or "image". In a similar way, it would be odd if we renamed `TextureAtlasSprite` to `TextureAtlasImage`.

Reply:
Great argument, changing to TextureAtlasImage to match the existing UiImage.  Sorry if I was being argumentative, I didn't know what alternative you were proposing.  :)